packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1803,7 +1803,13 @@ class BackdropFilterLayer extends ContainerLayer {
   ///
   /// The [filter] property must be non-null before the compositing phase of the
   /// pipeline.
-  BackdropFilterLayer({ ui.ImageFilter? filter }) : _filter = filter;
+  /// The [blendMode] property will be defaulted to [BlendMode.srcOver] if it is
```

Comment:
That would make sense, but the 4 Layer classes around this all use this model of taking nullable parameters in the constructor and then mentioning that they must be set before compositing. I think this is due to the callers using a paradigm of:

```
  layer ??= MyLayerType();
  layer.foo = fooValue;
  layer.bar = barValue;
```

Reply:
What are the other 4 Layer classes that follow this pattern?

I understand the pattern for the `filter` property because it doesn't have an explicit default, but if we have an explicit default, why not expose it?

Reply:
I guess there are 2 things here. You asked why it couldn't be non-nullable.

There are no properties on any Layer class that I can find that are non-nullable. The "4" I mentioned were just due to scrolling a little up and down and seeing the 4 adjacent Layer classes do the same thing. But if you scroll through the entire file you will see that *ALL* properties on all Layer classes are nullable. I'm not going to list them.

On the other hand, you also asked if it could take a default value.

There are some properties that do get a default value in their constructor. That is true, and I should probably provide a default value here (even though other uses of BlendMode properties don't provide defaults - mostly it is Offset parameters that have a default of zero).

But, even the parameters that have a default in the constructor are nullable properties. For whatever reason, and I wasn't in the room when these code styles were created, it just seems to be a given that properties are nullable regardless of whether they have defaults or not. If you think that is not right, then that should probably be taken up separately. Until then, I'll follow the local patterns here. I'll give it a default, but leave the discussion of the reasons these properties are all nullable for another issue/fix...

Reply:
I looked a little further and found some other patterns, ignore my previous comment.

Reply:
Done.